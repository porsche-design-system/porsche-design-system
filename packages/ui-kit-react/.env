SKIP_PREFLIGHT_CHECK=true

// BROWSERSTACK_USER_NAME=porscheuitechnicaluser
// BROWSERSTACK_ACCESS_KEY=4z6yJpVyVNHkz5yqAuSG

BROWSERSTACK_USER_NAME=oliverhook1
BROWSERSTACK_ACCESS_KEY=auteTfP2qWeEfMCehBg2
