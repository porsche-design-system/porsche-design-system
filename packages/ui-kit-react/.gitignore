# Build
/dist/

# Logs
*.log
npm-debug.log*
yarn-debug.log*
yarn-error.log*

# Dependency directories
node_modules/

# Optional npm cache directory
.npm

# Yarn Integrity file
.yarn-integrity

# dotenv environment variables file
.env

# next.js build output
.next

# operating system files
.DS_Store
Thumbs.db
