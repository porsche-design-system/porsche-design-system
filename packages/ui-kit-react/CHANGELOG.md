# Porsche UI Kit React

## [1.0.0-alpha.1] (2019-08-21)

### Features

* Initial release