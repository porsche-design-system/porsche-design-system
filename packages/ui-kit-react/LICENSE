SEE LICENSE IN https://ui.porsche.com
