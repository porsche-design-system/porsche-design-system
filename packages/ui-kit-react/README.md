# stencil-ds-react-template

This is an example repo of building plugins.

## Step 1.

- Update the `package.json` to have the correct package name for this repo.
- Replace `component-library` under `dependencies` with your core stencil package name.

## Step 2.

- Build your core stencil package.

## Step 3.

- Run build on this package.
