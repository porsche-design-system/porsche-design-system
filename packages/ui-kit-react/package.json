{
  "name": "ui-kit-react",
  "version": "0.0.0",
  "private": true,
  "scripts": {
    "start": "react-scripts start",
    "build": "cd projects/ui-kit-wrapper && yarn build",
    "test": "yarn test:vrt && yarn test:cbt",
    "test:vrt": "concurrently -k -s \"first\" -n \"server,vrt\" \"BROWSER=none yarn start\" \"wait-on -t 180000 http-get://localhost:3000 && ts-node --project 'tests/vrt/config/tsconfig.json' node_modules/.bin/jasmine --config='tests/vrt/config/jasmine.config.json'\"",
    "test:cbt": "concurrently -k -s \"first\" -n \"server,cbt\" \"BROWSER=none yarn start\" \"wait-on -t 180000 http-get://localhost:3000 && ts-node --project 'tests/cbt/config/tsconfig.json' node_modules/.bin/jasmine --config='tests/cbt/config/jasmine.config.json'\"",
    "clean": "rm -rf 'node_modules' && rm -rf 'yarn-error.log' && rm -rf 'dist'"
  },
  "dependencies": {
    "@porsche-design-system/components-react": "^1.0.0-rc.1"
  },
  "devDependencies": {
    "@porsche-ui/cross-browser-tester": "^1.0.2",
    "@porsche-ui/visual-regression-tester": "^1.0.0",
    "@types/jest": "24.0.23",
    "@types/node": "12.7.5",
    "@types/react": "16.9.2",
    "@types/react-dom": "16.9.0",
    "jasmine": "^3.5.0",
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "react-scripts": "3.2.0",
    "react-test-renderer": "^16.10.2",
    "typescript": "3.6.3"
  },
  "eslintConfig": {
    "extends": "react-app"
  },
  "license": "SEE LICENSE IN https://designsystem.porsche.com",
  "browserslist": {
    "production": [
      "last 2 version",
      ">1%",
      "not dead",
      "not ie < 11",
      "not op_mini all"
    ],
    "development": [
      "last 2 version",
      ">1%",
      "not dead",
      "not ie < 11",
      "not op_mini all"
    ]
  }
}
