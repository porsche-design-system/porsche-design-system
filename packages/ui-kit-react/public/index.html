<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8">
  <title>Porsche UI Kit React</title>
  <base href="/">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <link rel="icon" type="image/x-icon" href="%PUBLIC_URL%/favicon.ico">
</head>
<body>
  <div id="root"></div>
  <script type="text/javascript">
    document.addEventListener('DOMContentLoaded', function() {
      var html = document.querySelector('html');
      function callOnHydrated(cb) {
        setTimeout(function () {
          if (html.classList.contains('hydrated')) {
            return cb();
          }

          return callOnHydrated(cb);
        }, 1000);
      }

      callOnHydrated(function () {
        /* Why? To see which browser is used to create cross browser tests by BrowserStack Screenshot API */
        var txt = 'Browser CodeName: ' + navigator.appCodeName + '<br>';
        txt+= 'Browser Name: ' + navigator.appName + '<br>';
        txt+= 'Browser Version: ' + navigator.appVersion + '<br>';
        txt+= 'Platform: ' + navigator.platform + '<br>';
        txt+= 'User-agent header: ' + navigator.userAgent;
        document.getElementById('system-log').innerHTML=txt;

        window.setTimeout(function() {
          /* Why? To force a redraw after everything has loaded to ensure IE11 renders the same each time */
          document.body.style.display = 'none';
          document.body.offsetHeight;
          document.body.style.display = 'block';
        }, 5000);
      });
    });
  </script>s
</body>
</html>
