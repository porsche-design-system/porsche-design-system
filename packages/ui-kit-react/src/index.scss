@import "~@porsche-ui/ui-kit-js/dist/porsche-ui-kit/porsche-ui-kit.css";
@import "~@porsche-ui/utils/dist/component-overview.css";
