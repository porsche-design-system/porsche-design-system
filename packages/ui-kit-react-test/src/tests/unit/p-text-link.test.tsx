import React from 'react';
import { PTextLink }  from '@porscheui/ui-kit-react';
import renderer from 'react-test-renderer';


describe("Text Link component", () => {
  it('should render correctly', () => {
    const component = renderer.create(
      <PTextLink>Text Link</PTextLink>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

  });
});

