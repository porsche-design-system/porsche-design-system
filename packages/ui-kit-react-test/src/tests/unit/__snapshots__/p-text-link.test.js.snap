// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Link component should render correctly 1`] = `
<p-text-link>
  Text Link
</p-text-link>
`;
